gensim/models/doc2vec.py

```diff
@@ -636,6 +636,12 @@ def scan_vocab(self, documents, progress_per=10000, trim_rule=None):
         interval_start = default_timer() - 0.00001  # guard against next sample being identical
         interval_count = 0
         vocab = defaultdict(int)
+        try:
+            if isinstance(list(documents)[0].words, string_types):
```

Comment:
Many users won't be passing an indexable/all-in-memory corpus – so this will break for them.